exasol_transformers_extension/utils/bucketfs_operations.py

```diff
@@ -36,18 +40,24 @@ def create_bucketfs_location(
 
 def upload_model_files_to_bucketfs(
         tmpdir_name: str, model_path: Path,
-        bucketfs_location: AbstractBucketFSLocation) -> None:
-    for tmp_file_path in Path(tmpdir_name).iterdir():
-        upload_model_file(bucketfs_location, model_path, tmp_file_path, tmpdir_name)
+        bucketfs_location: AbstractBucketFSLocation) -> Path:
+    with tempfile.TemporaryFile() as fileobj:
+        create_tar_of_directory(Path(tmpdir_name), fileobj)
+        return upload_file(bucketfs_location, fileobj, model_path)
 
 
 @retry(wait=wait_fixed(2), stop=stop_after_attempt(10))
-def upload_model_file(bucketfs_location, model_path, tmp_file_path, tmpdir_name):
-    with open(tmp_file_path, mode='rb') as file:
-        bucketfs_path = PurePosixPath(
-            model_path, tmp_file_path.relative_to(tmpdir_name))
-        bucketfs_location.upload_fileobj_to_bucketfs(
-            file, str(bucketfs_path))
+def upload_file(bucketfs_location: AbstractBucketFSLocation, fileobj: BinaryIO, model_path: Path) -> Path:
+    fileobj.seek(0)
+    model_tar_file = model_path.with_suffix(".tar.gz")
+    bucketfs_location.upload_fileobj_to_bucketfs(fileobj, str(model_tar_file))
+    return model_tar_file
+
+
+def create_tar_of_directory(path: Path, fileobj: BinaryIO):
+    with tarfile.open(name="test.tar.gz", mode="w|gz", fileobj=fileobj) as tar:
```

Comment:
maybe rename the test.tar.gz ? this is not a test after all

Reply:
good catch, actually don't know why I wrote test.tar.gz, maybe copy and paste from a test where I tried this first